### PR TITLE
Upgrade to latest rocksdb version in tikv/rust-rocksdb, and deal with api changes

### PR DESCRIPTION
**Pull request by thefallentree**

### CODE REVIEW

**Comment by thefallentree** [comment]
> build is still failing

I think test_full_node_basic_flow is flaky starting recently, I'm re-running it to verify

File: storage/libradb/src/state_store/mod.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ impl StateStore {
         let mut iter = self
             .db
             .iter::<JellyfishMerkleNodeSchema>(Default::default())?;
-        iter.seek_to_first();
+        iter.seek_to_first().unwrap();
```
Comment (wqfish) [inline storage/libradb/src/state_store/mod.rs]: It doesn't really matter in tests but can you use `?` here for consistency.

Comment (thefallentree) [reply storage/libradb/src/state_store/mod.rs]: done

**Pull request closed without merging**